restorers/model/mprnet/attention.py

```diff
@@ -0,0 +1,177 @@
+from typing import Optional, Type
+
+import tensorflow as tf
+
+
+class ChannelAttentionLayer(tf.keras.layers.Layer):
+    def __init__(
+        self,
+        channels: int,
+        reduction: Optional[int] = 4,
+        use_bias: Optional[bool] = False,
+        name: str = "Channel Attention Layer",
+        *args,
+        **kwargs
+    ) -> None:
+        super().__init__(name=name, *args, **kwargs)
+
+        self.avg_pool = tf.keras.layers.AveragePooling2D((1, 1))
+        self.stem = tf.keras.Sequential(
+            [
+                tf.keras.layers.Conv2D(
+                    filters=int(channels // reduction),
+                    kernel_size=1,
+                    padding="same",
+                    use_bias=use_bias,
+                ),
+                tf.keras.layers.ReLU(),
+                tf.keras.layers.Conv2D(
+                    filters=channels,
+                    kernel_size=1,
+                    padding="same",
+                    use_bias=use_bias,
+                ),
+                tf.keras.layers.Activation("sigmoid"),
+            ],
+        )
+
+    def call(self, inputs: tf.Tensor, training: Optional[bool] = None) -> tf.Tensor:
+        x = self.avg_pool(inputs)
+        x = self.stem(x)
+        return inputs * x
+
+    def get_config(self) -> dict:
+        config = super().get_config()
+        config.update(
+            {
+                "channels": self.channels,
+                "reduction": self.reduction,
+                "use_bias": self.use_bias,
+            }
+        )
+        return config
```

Comment:
Hi @SauravMaheshkar
I noticed that @Samarendra109 has already implemented a simplified basic variant of this block [here](https://github.com/soumik12345/restorers/blob/main/restorers/model/nafnet/nafblock.py#L33). Do you think you can refactor your implementation of `ChannelAttentionLayer` as a subclass of the `SimplifiedChannelAttention` class? That way it would reduce logic redundancy

Reply:
Rather than subclassing, I'd suggest making our custom `ChannelAttention` module and then having `SimplifiedChannelAttention` and `ChannelAttentionLayer` both subclass that. Because `ChannelAttentionLayer` has more convolution layers than the `SimplifiedChannelAttention` module, subclassing won't be ideal. This could also be a great way to have a conversation about API design about the `restorers` library.

Reply:
Yes, we have to come up with a clean design approach for blocks that are being reused across modules.